src/test/java/org/fcrepo/client/integration/FcrepoTransactionIT.java

```diff
@@ -0,0 +1,161 @@
+        }
+
+        try (final var response = client.transaction().commit(location).perform()) {
+            assertEquals(NO_CONTENT.getStatusCode(), response.getStatusCode());
+        }
+    }
+
+    @Test
+    public void testStartTransactionalClient() throws Exception {
+        // the same as testTransactionCommit but using startTransactionalClient
+        try (final var transactionalClient = client.transaction().startTransactionalClient(new URI(SERVER_ADDRESS))) {
+            final var txURI = transactionalClient.getTransactionURI();
+
+            final var container = UUID.randomUUID().toString();
+            try (final var response = transactionalClient.put(new URI(SERVER_ADDRESS + container)).perform()) {
+                assertEquals(CREATED.getStatusCode(), response.getStatusCode());
+                assertTrue(response.getTransactionUri().isPresent());
+                assertEquals(txURI.asString(), response.getTransactionUri().get().asString());
+            }
+
+            try (final var response = client.transaction().commit(txURI).perform()) {
+                assertEquals(NO_CONTENT.getStatusCode(), response.getStatusCode());
+            }
+        }
+    }
+
+    @Test
+    public void testTransactionKeepAlive() throws Exception {
+        final String expiry;
+        final FcrepoResponse.TransactionURI location;
+
+        try (final var response = client.transaction().start(new URI(SERVER_ADDRESS)).perform()) {
+            assertEquals(CREATED.getStatusCode(), response.getStatusCode());
+
+            location = response.getTransactionUri().orElseThrow(() -> new IllegalStateException("No tx found"));
+            // the initial transaction currently returns Expires rather than Atomic-Expires
+            expiry = response.getHeaderValue("Expires");
+        }
+
+        try (final var response = client.transaction().keepAlive(location).perform()) {
+            assertEquals(NO_CONTENT.getStatusCode(), response.getStatusCode());
+
+            final var expiryFromStatus = response.getHeaderValue(FedoraHeaderConstants.ATOMIC_EXPIRES);
+            assertNotNull(expiryFromStatus);
+
+            final var initialExpiration = Instant.from(FORMATTER.parse(expiry));
+            final var updatedExpiration = Instant.from(FORMATTER.parse(expiryFromStatus));
+            assertTrue(initialExpiration.isBefore(updatedExpiration));
+        }
+    }
+
+    @Test
+    public void testTransactionStatus() throws Exception {
+        // the initial transaction currently returns Expires rather than Atomic-Expires, update if changed
+        final String expiry;
+        final FcrepoResponse.TransactionURI location;
+
+        try (final var response = client.transaction().start(new URI(SERVER_ADDRESS)).perform()) {
+            assertEquals(CREATED.getStatusCode(), response.getStatusCode());
+
+            location = response.getTransactionUri().orElseThrow(() -> new IllegalStateException("No tx found"));
```

Comment:
I'm not totally against the optional, but from a expectations perspective, I think it'd make sense to have `getTransactionUri` return `TransactionURI` directly. 

With the optional it is shorter to throw an error if there is no txUri. But if the user requested to start a tx and the response didn't have a txUri, it seems like that would indicate a server error of some sort, and should have already thrown an exception. Also, to get the uri right now, the user would have to call `response.getTransactionUri().get().get()` or similar with `orElseThrow`, which looks kind of weird.

I'm also trying to figure out what having a TransactionURI class is getting us versus passing back a URI. It only holds a single value, and doesn't seem to have any internal logic aside from calling URI.create.

Reply:
Yea I was hoping the TransactionURI would provide more value but I can't really say much has emerged. I was actually thinking about rolling it back before initially pushing the commit with it but went with it just to see how it would go. I'm still kind of on the fence about it solely from using it in the TransactionBuilder.

I can see what you're saying wrt the Optional as well - failures in the request would have already occurred by the time we're calling `getTransactionUri`. Originally I wanted the `start` to return something like TxAwareFcrepoResponse but didn't want to make another class for that single method.